javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java

```diff
@@ -64,6 +81,127 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
+                            if (bodyMap.containsKey(method.getMethodPageDetails().getSerializedItemName())) {
+                                Object items = bodyMap.get(method.getMethodPageDetails().getSerializedItemName());
+                                if (items instanceof List) {
+                                    List<Object> itemArray = (List<Object>) items;
+                                    if (itemArray.isEmpty()) {
+                                        methodBlock.line("Assertions.assertEquals(0, response.stream().count());");
+                                    } else {
+                                        Object firstItem = itemArray.iterator().next();
+                                        methodBlock.line("%s firstItem = %s;", responseType.getTypeArguments()[0], "response.iterator().next()");
+                                        writeModelAssertion(methodBlock, nodeVisitor, responseType.getTypeArguments()[0], responseType.getTypeArguments()[0], firstItem, "firstItem");
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    writeModelAssertion(methodBlock, nodeVisitor, returnType, returnType, response.getBody(), "response");
+                }
+            } else {
+                methodBlock.line("Assertions.assertNotNull(response);");
+            }
+        };
+    }
+
+    private void addClientModelImports(IType type) {
+        ClientModel clientModel = null;
+        if (type instanceof GenericType) {
+            GenericType responseType = (GenericType) type;
+            if (PagedIterable.class.getSimpleName().equals(responseType.getName())
+                    && ClientModelUtil.isClientModel(responseType.getTypeArguments()[0])) {
+                clientModel = ClientModelUtil.getClientModel(((ClassType) responseType.getTypeArguments()[0]).getName());
+            }
+        } else if (ClientModelUtil.isClientModel(type)) {
+            clientModel = ClientModelUtil.getClientModel(((ClassType) type).getName());
+        }
+        if (clientModel != null) {
+            clientModel.addImportsTo(this.imports, JavaSettings.getInstance());
+        }
+    }
+
+    private void writeModelAssertion(JavaBlock methodBlock, ModelExampleWriter.ExampleNodeModelInitializationVisitor nodeVisitor,
+                                     IType modelClientType, IType modelWireType, Object modelValue, String modelReference) {
+        if (modelValue != null) {
+            if (modelClientType instanceof MapType || modelClientType instanceof ArrayType) {
+                methodBlock.line("Assertions.assertNotNull(%s);", modelReference);
+            } else if ((!ClientModelUtil.isClientModel(modelClientType) && (!(modelClientType instanceof ListType)))
+                    || modelClientType instanceof EnumType) {
```

Comment:
Should we re-arrange the order of if-else?

1. client model
2. list
3. other that can use `assertEquals` (may whitelist them)
4. rest be `assertNotNull`

Reply:
Yeah, we should do that. We don't want to falsely include those which can't be compared by equals.